app/api/user.py

```diff
@@ -91,6 +120,100 @@ def profile():
     user = current_user["_id"]
     return(str({"login as ": user})), 200
 
+@bp.route('/reset_pass/<string:id>', methods=['PUT'])
+def pass_Reset(id):
+    log_username = request.json.get("username", None)
+    email =request.json.get("email", None)
+    password = request.json.get("password", None)
+    new_password = request.json.get("new_password1", None)
+    retype_new_password = request.json.get("retype_new_password", None)
+
+
+    hash = pbkdf2_sha256.hash("password")
+    if not log_username:
+        return jsonify(msg="Missing username parameter"), 400
+    if not password:
+        return jsonify(msg="Missing password parameter"), 400
+    if password == new_password :
+        return jsonify(msg="should be diffrent from old password")
+    if new_password != retype_new_password :
+        return jsonify(msg="password done not match"), 400
+
+    E_mail = mongo.db.users.find_one({"email": email})
+    if E_mail is None:
+        return jsonify(msg = "email sone not exist")
+
+    is_user = mongo.db.users.find_one({"username": log_username })
+    if is_user is None:
+        return jsonify(msg="username doesn't exists"), 400
+
+    if not pbkdf2_sha256.verify(password, is_user["password"]):
+        return jsonify(msg="password is wrong"), 400
+
+
+    update_json = {}
+    if new_password is not None:
+         update_json["password"] = hash
+    ret = mongo.db.users.update({
+       "_id": ObjectId(id)
+         }, {
+       "$set": update_json
+         }, upsert=False)
+
+    if ret is not None:
+        msg = Message('Password Change Alert', sender = 'mohit_saini@excellencetechnologies.info', recipients = [email] )
+        msg.body = "Warning - Password Chnage Request Found  " 
+        mail.send(msg)  
+
+    return(ret)
+
+
+@bp.route('/forgot_pass', methods=['GET']) 
+def pass_Forgot(): 
+    
+    email = request.json.get("email", None)
+    if not email:
+        return jsonify(msg="Missing username parameter"), 400
+
+    global content
+    content = email
+
+    is_mail= mongo.db.users.find_one({"email":email})
+    if is_mail is not None:
+        msg = Message('Password Forgot Request', sender = 'mohit_saini@excellencetechnologies.info', recipients =[content] )
+        msg.body = "Request for New Password Is Accepted - Open The Link And Get A New Random Password "
+        msg.html = "(<a href ='http://127.0.0.1:5000/set_pass'>Click Here To Chnage Password</a>)"
```

Comment:
here should base64 encrypted email also in api parameter.. ?email= like this.... then only you can get the email of the user in set_pass api for that we need we need to change password.....